### PR TITLE
SDK Download command enhancement

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23845

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: demisto_sdk/commands/download/downloader.py [hunk 1]
```diff
@@ -216,23 +299,23 @@ def build_custom_content(self) -> None:
...
-        # Build custom content files
-        custom_content_file_paths: list = get_child_files(self.custom_content_temp_dir)
-        custom_content_objects: list = list()
-        for custom_content_file_path in custom_content_file_paths:
-            custom_content_objects.append(self.build_custom_content_object(custom_content_file_path))
-
         for input_file_name in self.input_files:
             input_file_exist_in_cc: bool = False
-            for custom_content_object in custom_content_objects:
+            for custom_content_object in self.get_custom_content_objects():
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/download/downloader.py:304]: If the `-a` flag is up this process is happening twice? It is happening also on line 185 any chance we can reuse it?

File: demisto_sdk/commands/format/format_module.py [hunk 2]
```diff
@@ -35,7 +35,7 @@
 }
 
 
-def format_manager(input: str = None, output: str = None, from_version: str = None, no_validate: str = None):
+def format_manager(input: str = None, output: str = None, from_version: str = None, no_validate: bool = None):
```
[the remark this answers is not in the document]
Comment (guyfreund) [reply demisto_sdk/commands/format/format_module.py:38]: Yes, it's a mistake. It is passed as an argument to all update objects, checked all of them.

File: demisto_sdk/commands/download/downloader.py [hunk 3]
```diff
@@ -39,52 +44,153 @@ class Downloader:
     Attributes:
         output_pack_path (str): The path of the output pack to download custom content to
         input_files (list): The list of custom content files names to download
-        insecure (bool): Indicates whether to you insecure connection or not
         force (bool): Indicates whether to merge existing files or not
+        insecure (bool): Indicates whether to you insecure connection or not
         log_verbose (bool): Indicates whether to use verbose logs or not
         client (Demisto client): The Demisto client to make API calls
+        list_files (bool): Indicates whether to print the list of available custom content files and exit or not
```
[the remark this answers is not in the document]
Comment (guyfreund) [reply demisto_sdk/commands/download/downloader.py:51]: If -list-files is provided we exit the command, if not we continue...

File: demisto_sdk/commands/download/downloader.py (hunk 1 above)
Comment (guyfreund) [reply demisto_sdk/commands/download/downloader.py:304]: Yes, refactored

**Pull request merged**